### PR TITLE
keep eval related computations and data on gpu

### PR DESCRIPTION
**Pull request by eordentlich**

### CODE REVIEW

**Comment by wbo4958** [comment]
Overall, the idea looks good to me. However, seems we can do further enhancement if we ensure the predicted/transformed data is on the GPU before evaluating. like [here](https://github.com/NVIDIA/spark-rapids-ml/blob/branch-24.02/python/src/spark_rapids_ml/regression.py#L746) and [here](https://github.com/NVIDIA/spark-rapids-ml/blob/branch-24.02/python/src/spark_rapids_ml/classification.py#L1474-L1500)

**Comment by eordentlich** [comment]
That is exactly what is being done in the PR here:  https://github.com/NVIDIA/spark-rapids-ml/pull/587/files#diff-5bffca677cda02544897eeac447779db91115ec59138bc10c6115e56ed123692R1476-R1482
But only for logistic regression so far.  Needs to be extended to the other crossvalidator supported models like random forest and linear regression.

**Comment by eordentlich** [comment]
build

**Comment by eordentlich** [comment]
build

File: python/src/spark_rapids_ml/regression.py [hunk 1]
```diff
@@ -741,9 +746,16 @@ def _construct_lr() -> CumlT:
 
             return lrs
 
-        def _predict(lr: CumlT, pdf: TransformInputType) -> pd.Series:
-            ret = lr.predict(pdf)
-            return pd.Series(ret)
+        if eval_metric_info:
+
+            def _predict(lr: CumlT, pdf: TransformInputType) -> pd.Series:
```
Comment (lijinf2) [inline python/src/spark_rapids_ml/regression.py:751]: nit: Is the return type CPU pd.Series or a GPU type (e.g. cp.ndarray)?

Comment (eordentlich) [reply python/src/spark_rapids_ml/regression.py:751]: good catch, should be cp.ndarray

**Comment by eordentlich** [comment]
build

**Review by lijinf2 — APPROVED** [review]

**Comment by eordentlich** [comment]
build

**Pull request merged**